sui_programmability/examples/basics/sources/Sandwich.move

```diff
@@ -22,8 +22,17 @@ module Basics::Sandwich {
         id: VersionedID,
     }
 
-    /// Address selling ham, bread, etc
-    const GROCERY: address = @0x0;
+    // This Capability allows the owner to withdraw profits
+    struct GroceryOwnerCapability has key {
+        id: VersionedID
```

Comment:
I think you will need a way to match a cap with the grocery, otherwise someone with a entirely new cap could redeem all grocery coins.

Reply:
Cap is only created on module init, so for one publish of the module can be only one cap.

Reply:
For usability sake it is nice to have id stored inside, but I was trying to keep simplicity of this example. The goal is to show off composability, and I’ve already added too much.

Reply:
Oh I see!
I would be curious to understand better the trade-offs between the two modes:
1. The module creates singletons, and if anyone wants to create a new store, they publish a new package
2. vs The module is capable of creating multiple stores, owned by different people.

Reply:
Yup, that’s the tradeoff we need to research. I always vote for reusability to max. But in some cases solutions are too specific and there’s no point in making them generic.